components/Chrome.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from 'react';
+import Image from 'next/image';
+import cx from 'classnames';
+import ProfilePic from '../public/profile.jpg';
+import FloatingNav from './FloatingNav';
+import Footer from './Footer';
+import Link from 'next/link';
+import LanguageToggler from './LanguageToggler';
+import { Lang } from '../lib/types';
+import { LanguageContext } from '../lib/LanguageContext';
+
+interface Props {
+  page: string;
+  smallFooter?: boolean;
+  children: React.ReactNode;
+}
+
+const Chrome: React.FC<Props> = ({ page, smallFooter, children }) => {
+  const [navOpen, setNavOpen] = useState(false);
+  const [language, setLanguage] = useState<Lang | null>(null);
+
+  useEffect(() => {
+    const detectedLang = localStorage.getItem('language');
+    if (!detectedLang || (detectedLang !== 'en' && detectedLang !== 'es')) {
+      if (window.navigator.language.startsWith('es')) {
+        localStorage.setItem('language', 'es');
+        setLanguage('es');
+      } else {
+        localStorage.setItem('language', 'en');
+        setLanguage('en');
+      }
+    } else {
+      localStorage.setItem('language', detectedLang);
+      setLanguage(detectedLang);
+    }
+  }, []);
+
+  if (!language) {
+    return (
+      <div className="h-screen flex justify-center items-center font-inter text-3xl text-gray-500 animate-pulse">
+        loading...
+      </div>
+    );
+  }
+
+  return (
+    <LanguageContext.Provider value={language}>
+      <div className="flex flex-col min-h-screen relative">
+        <div
+          className={cx(
+            'z-40 bg-black left-0 top-0 w-screen h-screen md-lg:hidden bg-opacity-70 fixed',
+            navOpen ? 'block' : 'hidden',
+          )}
+          onClick={() => setNavOpen(false)}
+        />
+        <nav
+          className={cx(
+            'fixed md-lg:hidden h-screen w-72 bg-white shadow-xl z-50 left-0 top-0 [transition:150ms] rounded-r-xl overflow-hidden',
+            navOpen ? 'ml-0' : '-ml-80',
+          )}
+        >
+          <button
+            className="absolute top-0 right-0 p-4 text-gray-300 hover:text-gray-400 transition duration-100"
+            onClick={() => setNavOpen(false)}
+          >
+            <i className="fa-solid fa-times text-2xl" />
+          </button>
+          <div className="p-6 mt-10 flex flex-col space-y-4">
+            {language === 'en' ? (
```

Comment:
hmmm... i'm not sure about this. it might be right, but with server rendering stuff, i'm a bit unclear wether the language would be available at build time. i think we need to only _dynamically generate on the server_ the _homepage_, all the rest of the pages, should be buildable ahead of time, and thus, they should have the correct language in the sidebar (and everywhere), and the only thing that changes is how the homepage links out to them.   we should have a little chat about this. i'm only mostly clear how this works.